lib/cadet_web/views/assessments_view.ex

```diff
@@ -120,7 +120,12 @@ defmodule CadetWeb.AssessmentsView do
         :mcq -> & &1.answer["choice_id"]
       end
 
-    transform_map_for_view(answer, %{answer: answer_getter, comment: :comment})
+    transform_map_for_view(answer, %{
+      answer: answer_getter,
+      comment: :comment,
+      xp: &((&1.xp || 0) + (&1.xp_adjustment || 0)),
```

Comment:
Did you check if this can be removed as well?

Reply:
Yeah, unfortunately I can't cos of `assessment_with_questions_and_answers/2`, we have `|> select([q, a], %{q | answer: a})` after left join. Unfortunately ecto makes it such that default can't even override it. It's a struct `Answer` with all attributes `nil` :((
```elixir
  def assessment_with_questions_and_answers(                                        
        assessment = %Assessment{id: id},                                           
        user = %User{role: role}                                                    
      ) do                                                                          
    if Timex.after?(Timex.now(), assessment.open_at) or role in @open_all_assessment_roles do
      answer_query =                                                                
        Answer                                                                      
        |> join(:inner, [a], s in assoc(a, :submission))                            
        |> where([_, s], s.student_id == ^user.id)                                  
                                                                                    
      questions =                                                                   
        Question                                                                    
        |> where(assessment_id: ^id)                                                
        |> join(:left, [q], a in subquery(answer_query), q.id == a.question_id)  
        |> select([q, a], %{q | answer: a})                                         
        |> order_by(:display_order)                                                 
        |> Repo.all()                                                               
                                                                                    
      assessment = Map.put(assessment, :questions, questions)                       
      {:ok, assessment}                                                             
    else                                                                            
      {:error, {:unauthorized, "Assessment not open"}}                              
    end                                                                             
  end
```